susemanager-utils/testing/docker/scripts/schema_idempotency_test_pgsql.py

```diff
@@ -0,0 +1,225 @@
+        changedFiles = []
+        files = json.loads(read(f))['files']
+        for file in filesDict:
+            if file['status'] == 'modified' or file['status'] == 'added':
+                changedFiles.append(file['filename'])
+    return(changedFiles)
+
+
+def get_all_files_since(version, schemaPath):
+    files = []
+    for element in os.listdir(schemaPath):
+        elementVer = re.search(
+            '^susemanager-schema-(\d+\.\d+[\d|.]*)-to-susemanager-schema-\d+\.\d+[\d|.]*$', element)
+        if elementVer:
+            # StrictVersion does not work as we have some migrations with version X.Y.W.Z
+            # https://hg.python.org/cpython/file/tip/Lib/distutils/version.py#l93
+            if DV.LooseVersion(elementVer.group(1)) >= DV.LooseVersion(version):
+                basePath = schemaPath + '/' + element
+                for sqlFile in os.listdir(basePath):
+                    if re.search('^.*\.(sql|sql\.postgresql|sql\.oracle)$', sqlFile):
+                        files.append(basePath + '/' + sqlFile)
+    return(files)
+
+
+def find_latest_version(schemaPath):
+    latestVersion = '0.0'
+    for element in os.listdir(schemaPath):
+        elementVer = re.search(
+            '^susemanager-schema-\d+\.\d+[\d|.]*-to-susemanager-schema-(\d+\.\d+[\d|.]*)$', element)
+        if elementVer:
+            if DV.LooseVersion(elementVer.group(1)) > DV.LooseVersion(latestVersion):
+                latestVersion = elementVer.group(1)
+    return(latestVersion)
+
+
+def find_next_version(schemaPath):
+    lastVersion = find_latest_version(schemaPath)
+    return(lastVersion.rsplit('.', 1)[0] + '.' + str(int(lastVersion.rsplit('.', 1)[1])+1))
+
+
+def create_fake_migration_path(schemaPath, newVersion, prFile=None, version=None):
+    if prFile:
+        print("Creating migration path with the scripts from the PR")
+        files = get_all_files_from_pr(prFile, schemaPath)
+    if version:
+        print("Creating migration path with all scripts since %s" % version)
+        files = get_all_files_since(version, schemaPath)
+    lastVersion = find_latest_version(schemaPath)
+    fakePath = schemaPath + \
+        '/susemanager-schema-%s-to-susemanager-schema-%s/' % (
+            lastVersion, newVersion)
+    print("Creating: " + fakePath)
+    os.mkdir(fakePath)
+    for migrationFile in files:
+        shutil.copy(migrationFile, fakePath)
+    #run_command(sys.path[0] + "/build-schema.sh")
+
+
+def run_command(command):
+    print(command)
+    try:
+        text = subprocess.check_call(command, shell=True)
+        return(True)
+    except subprocess.CalledProcessError as e:
+        print("Return code was %s " % e.returncode)
+        return(False)
+    except Exception as e:
+        raise
+
+
+def manage_postgresql(action):
+    actions = ['start', 'stop']
+    if action not in actions:
+        raise "Invalid action for the PostgreSQL service"
+    if file_exists("/usr/lib/postgresql-init"):
```

Comment:
In what case does this not exist? Can't we always use `rcpostgresql`?

Reply:
This is basically a copy paste of what we do at `schema_migration_test_pgsql-31to40.sh` (we use bash there), so I guess there is a good reason... or maybe just a historic one :-D

@mcalmer?

Reply:
Maybe this was added with postgres 9.6 and the old rcpostgresql script do not work anymore.
So if this script exists, use it, otherwise try rcpostgresql command.

If you work for Uyuni only you can use /usr/lib/postgresql-init exclusivly

Reply:
Well, in fact the base OS used at the container is SLE12 using PostgreSQL 9.6.

So this means we can safely use `/usr/lib/postgresql-init` only from now on, as both Uyuni and future SUSE Manager will use >= 9.6.

Thanks for the clarification @mcalmer, I will fix both tests.